manifests/base/mc-fleet-deployment.yaml

```
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: repo-server
+    app.kubernetes.io/name: argocd-repo-server
+    app.kubernetes.io/part-of: argocd
+  name: argocd-repo-server
+  namespace: argocd
+spec:
+  containers:
+  - name: argocd-repo-server
+    volumeMounts:
+    - mountPath: /usr/local/bin/konfigure
+      name: giantswarm-bin
+      subPath: konfigure
+  initContainers:
+  - args:
+    - --vault-address={{ .Values.Installation.V1.Auth.Vault.Address }}
```

How do we get `vault-address` here?

---

It's different per installation.

---

One idea that comes to my mind is getting that as `envFrom.configMapRef` from some CM created during installation process.

@corest do you have ideas?

---

How do you feel about creating `vault` Service in `giantswarm` namespace? It could redirect to the actual vault address from installations but saves us all the extra `envFrom.configMapRef` magic. We'd get `vault.svc.cluster.local:443` everywhere and the flag could look like this:
```suggestion
    - --vault-address=vault.svc.cluster.local:443
```

---

I'm not sure if the CA will work with that.

---

How do you route that service to vault? Vault is running outside of the Kubernetes. Also as Pawel mentioned, CA will not work

---

Ah, missed the CA part. I'll use CM ref here.